src/life-calendar.js

```diff
@@ -292,16 +295,20 @@ function makeLifeCalendar(calendar, options) {
     // Periods
     var bgcolor = null;
     var stclass = null;
+    var innerHTML = null;
     periods.forEach(period => {
       if ( !(dateEnd<=period["start"] || dateStart>=period["end"]) ) {
         info["periods"].push(period);
+        if (period["icon"] && innerHTML===null){ innerHTML = period["icon"]; }
         if (period["backgroundColor"] && bgcolor===null){ bgcolor = period["backgroundColor"]; }
         if (period["className"] && stclass===null){ stclass = period["className"]; }
       }
     });
+    if (innerHTML){ itemspan.innerHTML = innerHTML; }
```

Comment:
I am not very sure about icon for events but I will consider it over this weekend.
This line will overwrite "event icon" with "period icon" and I don't think it's a good idea. "Event icon" should be prioritized and "period icon" is only allow when there is no event in that square.

Reply:
Regarding the usage, it was mainly when I was trying to log my travels I wanted to have flags for different countries I was staying in while traveling for longer periods.

And regarding the overwriting bit, I agree that event icon should be prioritized over period icon. I will make changes for that. But for background color and styling, shouldn't in that case as well event be prioritized over period? Should I also make changes to reflect that?

Thank you.

Reply:
Periods and events already has different background color and styling so there is no need to prioritize one or the other so you don't need to change any code.

Icon is supposed to just for events but if we're gonna do icon for periods to the only reasonable way I can think of is share the icon between period and event so you only need to change that line of code I guess.

I will do more careful review later.

Reply:
Ohh yes, I get it. I have addressed icon priority in the latest commit. Thanks in advance for taking a look.